test/app-luatest/override_misc_test.lua

```diff
@@ -52,3 +52,21 @@ g.test_initializing = function(g)
     local exp = expected_output('socket')
     t.assert_equals(res, exp)
 end
+
+g.test_no_package_loaded = function(g)
```

Comment:
I suggest to make test-case a little bit more interesting (or add another one):
After successful require, add our module to `no_package_loaded` and require it again to be sure that it will be removed from `package.loaded`.

Reply:
AFAIU you meant adding into `no_package_loaded` from outside of the module. It is not supposed to be used this way. In fact, it is not needed if you anyway have full control: add or remove anything from `package.loaded`.

This field is only needed to control `require()` behavior from inside the module. I don't know, whether it is good to hold a behavior that is not supposed to be used anyhow.

Reply:
OK, I agree, that it may be stayed "undefined" (mean no guarantees) for users.